### ISSUE TITLE
substr() on purchase button conflicts with custom edd_get_purchase_link() parameters

### ISSUE THREAD

**Issue by SDavisMedia** [easydigitaldownloads/edd-free-downloads#166]
In an attempt to restructure to purchase form, this like is used: https://github.com/easydigitaldownloads/edd-free-downloads/blob/4300e757c31c2f18892d1e46bdf99164a848e398/includes/filters.php#L94

The problem is anyone using `edd_get_purchase_link()` can have custom values for the parameters like so:

```php
//edd_price(get_the_ID());
echo edd_get_purchase_link( array(
	'download_id' => get_the_ID(),
	'class' => 'btn btn-sm btn-solid base_clr_bg item-link-buy',
	'text' => '<span class="price-title"><span class="icon icon-shopping-18"></span>'.$button_text.'</span>'.'<span class="price-value">'.edd_currency_filter(edd_get_download_price(get_the_ID())).'</span>',
	'price' => (bool)false
));
```

With that example, Free Downloads will cut the `text` value off at `-` in the `price-title` class and return the rest of the string. Here's the result on the front end:
<img width="206" alt="screenshot 2017-06-22 10 00 56" src="https://user-images.githubusercontent.com/2359131/27446243-b37503e4-5731-11e7-9e6c-947008c88b9e.png">

Free Downloads will need to take a different approach for rebuilding the purchase form.

**Comment by SDavisMedia** [comment]
Ticket: https://secure.helpscout.net/conversation/384135307/61413/?folderId=321418

**Comment by ghost1227** [comment]
Well that's fun! I'll get right on it!

**Comment by ghost1227** [comment]
@SDavisMedia I've been trying to find a better way to do this, but nothing is standing out. Got any thoughts?

**Comment by pippinsplugins** [comment]
@ghost1227 I'm not sure I follow what the substr is doing here. Why is it needed?

**Comment by ghost1227** [comment]
@pippinsplugins With the way the button text is output by EDD, it includes the purchase price. Since this code is applied to free products only, by default you end up with a purchase button that says '$0.00 - Free' (or similar). The original purpose of the substr was to strip out that price.

**Comment by ghost1227** [comment]
However, as @SDavisMedia said, that method doesn't take into account non-standard forms.

**Comment by SDavisMedia** [comment]
In EDD 2.8, all of the individual components of the purchase button text will be wrapped in HTML (another reason why solid HTML is so important). https://github.com/easydigitaldownloads/easy-digital-downloads/issues/5657

With that, you may be able to ditch all of this logic and rely directly on CSS. 

```css
.edd-free-download .edd-purchase-button-free,
.edd-free-download .edd-purchase-button-separator {
    display: none;
}
````

**Comment by ghost1227** [comment]
@SDavisMedia EDD core is on 2.8.1, any word on when this change will happen? I'd love to be able to rely solely on CSS in the future...

**Comment by SDavisMedia** [comment]
@ghost1227 unfortunately, this enhancement was reverted in 2.8.1. :/ https://github.com/easydigitaldownloads/easy-digital-downloads/issues/5900 

I agree that a CSS solution would be best. And @cklosowski says he's going to try to work up a different approach that still includes the new HTML. So perhaps this can wait until after that.

**Comment by cklosowski** [comment]
Yeah it'll have to wait. I need to build the platform to call all the same previous hooks and filters, yet retain the markup, so going to take some trickery.

**Comment by ghost1227** [comment]
Roger that!

**Comment by cklosowski** [comment]
Punting from 2.2.0, the rewrite of the EDD Core function is not done yet and will make this a _far_ less 'hacked' approach.
